frontend/src/lib/components/apps/editor/AppEditorHeader.svelte

```diff
@@ -57,13 +57,35 @@
 
 	export let policy: Policy
 
-	const { app, summary, mode, breakpoint, appPath, jobs, staticExporter, errorByComponent } =
-		getContext<AppEditorContext>('AppEditorContext')
+	const {
+		app,
+		summary,
+		mode,
+		breakpoint,
+		appPath,
+		jobs,
+		staticExporter,
+		errorByComponent,
+		eventBus
+	} = getContext<AppEditorContext>('AppEditorContext')
+
 	const loading = {
 		publish: false,
 		save: false
 	}
 
+	$: if ($eventBus.length > 0 && $eventBus[0].name === 'debug-runs') {
+		const { data: componentId } = $eventBus[0]
+		jobsDrawerOpen = true
+
+		const job = $jobs.find((job) => job.component === componentId)
+		if (job) {
+			selectedJobId = job.job
+		}
+
+		eventBus.update((events) => events.slice(1))
+	}
```

Comment:
An event bus seems quite complex compared to putting an:
`openDebugRun(componentId)`

any strong reasons for it ? What else do you see having there in the future ?

Reply:
It's a pattern that is already used in the app editor, for example opening the connect input menu.
I would only use one pattern for theses kind of cross-component communication

Reply:
I still don't understand the benefit. This adds many layers of indirection and bug possibilities for no apparent benefits.

The alternative would be to literally just pass the function `openDebugRun(componentId)`  in a global context which is a pattern we use almost everywhere. That function can be passed up using component let bindings. It's a lot cleaner

Reply:
Done